pkg/models/repository.go

```diff
@@ -27,12 +28,41 @@ func (r *Repository) BeforeCreate(tx *gorm.DB) (err error) {
 	if err := r.Base.BeforeCreate(tx); err != nil {
 		return err
 	}
+	if err := r.validate(); err != nil {
```

Comment:
Question,  so when a user enters a URL that has " https://some.url.com " 
Would this "validate" throw an error? 

As I understand it your CleanupURL function would remove the above leading/trailing whitespace.
So would you not want to first check for empty string, and then run cleanupURL, then run the stringContainsWhitespace (if at all) to check for internal whitespace? 

I'm probably misunderstanding this.

Reply:
stringContainsWhitespace also trims the leading/trailing whitespace

Reply:
ahh.. I see the strings.TrimSpace(s), this makes more sense now.

Reply:
An optional change you could do would be to change the name from stringContainsWhitespace to something like: 
stringContainsInternalWhitespace?

Reply:
I'm bad with names..